### PR TITLE
Move to Phpspec's php-diff to show line numbers

### PR DESCRIPTION
**Pull request by cdaguerre**

### CODE REVIEW

File: Symfony/CS/Fixer.php [hunk 1]
```diff
@@ -343,18 +337,23 @@ public static function getLevelAsString(FixerInterface $fixer)
         return 'symfony';
     }
 
-    protected function stringDiff($old, $new)
+    protected function stringDiff($old, $new, $type = 'unified')
     {
-        $diff = $this->diff->diff($old, $new);
+        $diff = new Diff(explode(PHP_EOL, $old), explode(PHP_EOL, $new));
```
Comment (GrahamCampbell) [inline Symfony/CS/Fixer.php]: bad idea

Comment (GrahamCampbell) [inline Symfony/CS/Fixer.php]: this will be broken on windows if the file used linux line endings

Comment (GrahamCampbell) [inline Symfony/CS/Fixer.php]: you should ALWAYS just explode on `\n`

**Comment by keradus** [comment]
I think that instead of replacing the Diff maker we should allow to use one of them. And even register it's own diff tool that comply with needed interface!

**Comment by GrahamCampbell** [comment]
What's the status here?

**Comment by hkdobrev** [comment]
Hi! 👋  Do you think this could be finished soon?

**Comment by keradus** [comment]
Sadly, closing as abandoned.

Currently, there is totally different code handling that, that we make it already easy to have different diff providers. See https://github.com/FriendsOfPHP/PHP-CS-Fixer/tree/master/src/Differ .
I think new differ (whatever it would be) could be added easily and then it will be configurable which differ to use.
Do you want to work on that @hkdobrev ?

**Pull request closed without merging**